modules/internal/ArrayViewRankChange.chpl

```diff
@@ -22,9 +22,337 @@
 // This module provides an array domain map class that is used to
 // represent rank change slices of another array via a domain.
 //
+// Throughout this file, 'down*' refers to the higher-dimensional
+// array/domain/distribution that's been rank-changed in this view,
+// while 'up*' refers to the lower-dimensional, user-facing view.
+//
+// TODOs for this file:
+//
+// * the dist, dom, and arr classes store a certain amount of
+//   duplicated information -- should these be focused in the higher
+//   classes (dist/dom) and referenced by lower ones (dom/dist)?
+//
+// * I don't think I've done anything to privatize the dist or dom
+//   classes... should I?
```

Comment:
There are several things about this file that assume that the domain to be rank-changed is rectangular. That's not strictly necessary (nor is it for slicing?). It'd be nice to either clearly say that currently the language has no plans to support rank-change views of non-rectangular arrays/domains or to say that supporting such things is a TODO.